ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/video/VideoView.kt

```diff
@@ -0,0 +1,528 @@
+    private var viewTreeObserverListening = false
+
+    init {
+        clipToPadding = true
+        clipChildren = true
+
+        addView(
+            texture,
+            LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT, Gravity.CENTER),
+        )
+
+        texture.surfaceTextureListener = object : TextureView.SurfaceTextureListener {
+            override fun onSurfaceTextureAvailable(st: SurfaceTexture, w: Int, h: Int) {
+                if (released) return
+                player.setSurface(Surface(st))
+                if (!viewTreeObserverListening) {
+                    viewTreeObserver.addOnGlobalLayoutListener(layoutListener)
+                    viewTreeObserverListening = true
+                }
+                // If we were already prepared before, just resume; otherwise prepare
+                if (prepared) {
+                    safeSeekTo(resumePosMs)
+                    if (resumePlayWhenReady) start()
+                    applySizing()
+                } else {
+                    prepareIfNeeded()
+                }
+            }
+
+            override fun onSurfaceTextureSizeChanged(st: SurfaceTexture, w: Int, h: Int) =
+                applySizing()
+
+            override fun onSurfaceTextureDestroyed(st: SurfaceTexture): Boolean {
+                if (released) return true
+                // snapshot play state & position to resume after recreation
+                resumePlayWhenReady = isPlaying
+                resumePosMs = currentPosition
+                // pause before losing surface to avoid state exceptions
+                if (prepared) {
+                    safely(execute = {
+                        player.pause()
+                    })
+                }
+                player.setSurface(null)
+                return true // we release the surface
+            }
+
+            override fun onSurfaceTextureUpdated(st: SurfaceTexture) {
+                // No-op
+            }
+        }
+
+        controller = MediaController(context).apply {
+            setMediaPlayer(this@TextureVideoView)
+            if (showControls) {
+                setAnchorView(this@TextureVideoView)
+                isEnabled = true
+            }
+        }
+        setOnClickListener {
+            if (showControls) {
+                controller?.show()
+            }
+        }
+    }
+
+    // Inside TextureVideoView.kt
+    class PlaybackState(val positionMs: Int, val playWhenReady: Boolean)
+
+    fun getPlaybackState(): PlaybackState {
+        // If not prepared yet, return last requested state (resumePosMs / resumePlayWhenReady)
+        val position = if (prepared) {
+            try {
+                player.currentPosition
+            } catch (_: Throwable) {
+                resumePosMs
+            }
+        } else {
+            resumePosMs
+        }
+        val play = if (prepared) {
+            (
```

Comment:
Hmm is this parentheses needed?

Reply:
Thanks. I'll make this change in the next PR. Missed this comment somehow

Reply:
Going into #2717